docs/RFCS/20230605_workload_level_index_recommendation.md

```diff
@@ -0,0 +1,348 @@
+larger set of candidates to find representative indexes to optimize the workload. In this way, since the exploring range 
+becomes much larger, the potential results should be better, but at the expense of time and space consuming. One idea to 
+deal with it is issue two level workload index recommendations. The first one is a lightweight one using the idea 
+collecting existing index recommendations as candidates running comparatively frequent, the other one using the idea 
+above running less often.
+
+### Extracting valuable index representatives
+
+Whether attaching all the storing columns to the shallowest leaf node is the optimal solution remains controversial. As 
+you can see in the example below:
+```
+              t
+              |   
+              i store (j, k)     
+         /         \
+j store (k)         k
+```
+The optimal solution should be indexes `(i, j) storing (k)` and `(i, k)`. If we assign storing columns `(j, k)`
+to the right leaf node since the left and right leaf nodes are in the same level, then the output will be indexes 
+`(i, j) storing (k)` and `(i, k) storing (j)`, which is a suboptimal solution. If we attach them to the left child, that 
+will lead to the optimal result. One potential answer for this question is that we can compare the storing parts in the 
+internal node with the path from the current node to its leaf nodes. But the statistics we need to keep while traversing 
+the tree is too overwhelming.
+
+If we prefer a lightweight method, we can convert this problem to deciding which child node to push down the storing 
+columns of one internal node so that we can merge the storing parts with the one in the child node. But there remain 
+some problems. If we push them down too shallowly, there may be more opportunities to share more storing parts among 
+indexes. Otherwise, we may add it to a very large leaf node index, which may make the optimization not that obvious 
+since visiting a large index is costly if we only want few columns.
+
+Another problem is whether we need to merging all the storing parts originally from different index candidates in one 
+tree. Think about the example as follows:
+```
+    t
+    |
+    i store (j); store (k)
+  /   \
+j       k
+```
+If we merge them together, the storing part becomes `(j, k)`, then there is no leaf node can cover them. The final 
+output of the algorithm will be indexes `(i, j) storing (k)` and `(i, k)`, or `(i, j)` and `(i, k) storing (j)`. 
+However, the optimal solution should be `(i, j)` and `(i, k)` since the storing parts from different original indexes 
+are covered by them respectively. 
+
+To deal with this problem, we need to process the storing part from multiple original indexes one by one, which could 
+increase the workload extremely. I suggest that we can keep a counter for the number of original indexes one storing 
+part of a node, if it exceeds some limit, we will merge them.
```

Comment:
I don't fully understand this suggestion. Maybe you can flesh it out a bit more?